File: 2.iOS界面开发（基础篇）/4.UINavigationController/4.单元作业/1.首页导航栏/1.首页导航栏/ViewController.m
//
//  ViewController.m
//  1.首页导航栏
//
//  Created by wd on 2017/5/27.
//  Copyright © 2017年 ditiy. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    
    //[self.navigationController.navigationBar setBarTintColor:[UIColor grayColor]];
    [self.navigationController.navigationBar setTintColor:[UIColor grayColor]];
    
    

    UIBarButtonItem *leftBarButtonItem = [[UIBarButtonItem alloc]initWithImage:[UIImage imageNamed:@"scan"] style:UIBarButtonItemStylePlain target:self action:@selector(leftBarButtonItemPressed:)];
    
    
    UIBarButtonItem *rightBarButtonItem = [[UIBarButtonItem alloc]initWithImage:[UIImage imageNamed:@"info"] style:UIBarButtonItemStylePlain target:self action:@selector(rightButtonItemPressed:)];
    
    self.navigationItem.leftBarButtonItem = leftBarButtonItem;
    
    self.navigationItem.rightBarButtonItem = rightBarButtonItem;
    
    
    
    UITextField *inputTextField = [[ UITextField alloc]initWithFrame:CGRectMake(0, 0, 250, 30)];
    inputTextField.borderStyle = UITextBorderStyleRoundedRect;
    
    
    inputTextField.textColor = [UIColor colorWithRed:67.0/255.0 green:67.0/255.0 blue:67.0/255.0 alpha:1];
    inputTextField.font = [UIFont systemFontOfSize:14.0];
    
    inputTextField.leftViewMode = UITextFieldViewModeAlways;
    inputTextField.attributedPlaceholder = [[NSAttributedString alloc]initWithString:@"在千万海外商品中搜索" attributes:@{NSFontAttributeName:[UIFont systemFontOfSize:15],NSForegroundColorAttributeName:[UIColor colorWithRed:67.0/255.0 green:67.0/255.0 blue:67.0/255.0 alpha:1]}];

    inputTextField.leftView = [[UIImageView alloc]initWithImage:[UIImage imageNamed:@"search"]];
    
    inputTextField.backgroundColor = [UIColor colorWithRed:233/252.0 green:234/255.0 blue:236/255.0 alpha:1.0];
    
    self.navigationItem.titleView = inputTextField;
    
    
    
}

-(void)leftBarButtonItemPressed:(UIBarButtonItem *)item{
    
    UIViewController *leftVC = [UIViewController new];
    
    leftVC.title = @"扫描";
    leftVC.view.backgroundColor = [UIColor whiteColor];
    
    [self.navigationController pushViewController:leftVC animated:YES];
}

-(void)rightButtonItemPressed:(UIBarButtonItem *)item{
    
    UIViewController *rightVC = [UIViewController new];
    
    rightVC.title = @"信息";
    rightVC.view.backgroundColor = [UIColor whiteColor];
    
    [self.navigationController pushViewController:rightVC animated:YES];
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: 2.iOS界面开发（基础篇）/4.UINavigationController/4.单元作业/作业批改/NavigationBar/NavigationBar/ViewController.swift
//
//  ViewController.swift
//  NavigationBar
//
//  Created by 胡诚真 on 2017/5/4.
//  Copyright © 2017年 胡诚真. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    let searchBar: UISearchBar = UISearchBar()
    override func viewDidLoad() {
        super.viewDidLoad()
        navigationItem.titleView = searchBar;
        searchBar.placeholder = "在千万海外商品中搜索"
        if let searchBarField = searchBar .value(forKey: "_searchField") as? UITextField {
            searchBarField.backgroundColor = UIColor(red: 234.0/255, green: 235.0/255, blue: 237.0/255, alpha: 1)
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func scan(_ sender: Any) {
        print("scan")
    }
    @IBAction func message(_ sender: Any) {
        print("message")
    }

    @IBAction func tapOnView(_ sender: Any) {
        searchBar.endEditing(true)
    }
}

